disperser/disperser.go

```diff
@@ -17,16 +17,43 @@ import (
 	gcommon "github.com/ethereum/go-ethereum/common"
 )
 
+// BlobStatus represents the status of a blob.
+// The status of a blob is updated as the blob is processed by the disperser.
+// The status of a blob can be queried by the client using the GetBlobStatus API.
+// Intermediate states are states that the blob can be in while being processed, and it can be updated to a differet state:
+// - PROCESSING
+// - DISPERSING
+// - CONFIRMED
+// Terminal states are states that will not be updated to a different state:
+// - FAILED
+// - FINALIZED
+// - INSUFFICIENT_SIGNATURES
+//
+// Note: this docstring and the enum ones below are copied from the disperser.proto,
+// which is the source of truth for BlobStatus.
 type BlobStatus uint
 
 // WARNING: THESE VALUES BECOME PART OF PERSISTENT SYSTEM STATE;
 // ALWAYS INSERT NEW ENUM VALUES AS THE LAST ELEMENT TO MAINTAIN COMPATIBILITY
 const (
+	// PROCESSING means that the blob is currently being processed by the disperser
 	Processing BlobStatus = iota
+	// CONFIRMED means that the blob has been dispersed to DA Nodes and the dispersed
+	// batch containing the blob has been confirmed onchain
 	Confirmed
+	// FAILED means that the blob has failed permanently (for reasons other than insufficient
+	// signatures, which is a separate state). This status is somewhat of a catch-all category,
+	// containg (but not necessarily exclusively as errors can be added in the future):
+	// - blob has expired
+	// - internal logic error while requesting encoder API 
+	// - blob retry has exceeded its limit while waiting for blob finalization after confirmation
 	Failed
+	// FINALIZED means that the block containing the blob's confirmation transaction has been finalized on Ethereum
 	Finalized
+	// INSUFFICIENT_SIGNATURES means that the confirmation threshold for the blob was not met
+	// for at least one quorum.
 	InsufficientSignatures
+	// DISPERSING means that the blob is currently being dispersed to DA Nodes and being confirmed onchain
```

Comment:
have not yet confirmed onchain

Reply:
I literally just copied these from disperser.proto. But agree, updated both: [69f0b50](https://github.com/Layr-Labs/eigenda/pull/828/commits/69f0b50ea521607dfdc96caf170e07936d1835e6)